### PR TITLE
Version file introduced and version set to 0.1.0

### PR DESCRIPTION
**Pull request by erikzenker**
This PR prepares the version for the next release

### CODE REVIEW

File: include/alpaka/version.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+* You should have received a copy of the GNU Lesser General Public License
+* along with alpaka.
+* If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define ALPAKA_VERSION_MAJOR 0
+#define ALPAKA_VERSION_MINOR 1
+#define ALPAKA_VERSION_PATCH 0
```
Comment (BenjaminW3) [inline include/alpaka/version.hpp]: Could you please add the lines you removed from alpaka.hpp? I find the `ALPAKA_VERSION` using the predef version macros really helpful. However, they should now use the major, minor and patch defines. You are missing the trailing newlin.

File: alpakaConfig.cmake [hunk 2]
```diff
@@ -527,17 +527,20 @@ ENDIF()
...
+string(REGEX MATCH "([0-9]+)" ALPAKA_VERSION_MINOR  ${ALPAKA_VERSION_MINOR_HPP})
+string(REGEX MATCH "([0-9]+)" ALPAKA_VERSION_PATCH  ${ALPAKA_VERSION_PATCH_HPP})
+
+set(PACKAGE_VERSION "${ALPAKA_VERSION_MAJOR}.${ALPAKA_VERSION_MINOR}.${ALPAKA_VERSION_PATCH}")
 
 #-------------------------------------------------------------------------------
 # Set return values.
 #-------------------------------------------------------------------------------
-SET(alpaka_VERSION "${_ALPAKA_VER_MAJOR}.${_ALPAKA_VER_MINOR}.${_ALPAKA_VER_PATCH}")
+set(alpaka_VERSION "${ALPAKA_VERSION_MAJOR}.${ALPAKA_VERSION_MINOR}.${ALPAKA_VERSION_PATCH}")
```
Comment (BenjaminW3) [inline alpakaConfig.cmake]: Could you please write the cmake functions uppercase as long as the other lines are also uppercase?

**Comment by BenjaminW3** [comment]
You will have to add the include of the version.hpp file into the alpaka.hpp file.

**Comment by erikzenker** [comment]
* Added the boost version number again. It is using now the major, minor, patch defines
* Added newline at the end of the version.hpp file
* Corrected cmake function the uppercase writing

**Pull request merged**